app/src/main/java/com/infomaniak/drive/ui/MainViewModel.kt

```diff
@@ -184,16 +184,24 @@ class MainViewModel(
         postCurrentFolder(FileController.getFileById(folderId, userDrive))
     }
 
-    fun createMultiSelectMediator(): MediatorLiveData<Pair<Int, Int>> {
-        return MediatorLiveData<Pair<Int, Int>>().apply { value = /*success*/0 to /*total*/0 }
+    fun createMultiSelectMediator(): MediatorLiveData<MultiSelectMediatorState> {
+        return MediatorLiveData<MultiSelectMediatorState>().apply {
+            value = MultiSelectMediatorState(numberOfSuccessfulActions = 0, totalOfActions = 0, errorCode = null)
+        }
     }
 
-    fun updateMultiSelectMediator(mediator: MediatorLiveData<Pair<Int, Int>>): (FileResult) -> Unit = { fileRequest ->
-        val total = mediator.value!!.second + 1
-        mediator.value = if (fileRequest.isSuccess) {
-            mediator.value!!.first + 1 to total
-        } else {
-            mediator.value!!.first to total
+    fun updateMultiSelectMediator(mediator: MediatorLiveData<MultiSelectMediatorState>): (FileResult) -> Unit = { fileRequest ->
+        var numberOfSuccessfulActions = mediator.value!!.numberOfSuccessfulActions
+        if (fileRequest.isSuccess) numberOfSuccessfulActions++
+
+        val totalOfActions = mediator.value!!.totalOfActions + 1
+
+        mediator.value = mediator.value?.let {
```

Comment:
Why are you doing `mediator.value?.let { ...` ?

Reply:
It was some oxbow code, I removed it.